pkg/doc/did/helpers.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package did
+
+import (
+	"fmt"
+)
+
+// LookupService returns the service from the given DIDDoc matching the given service type.
+func LookupService(didDoc *Doc, serviceType string) (*Service, error) {
+	const notFound = -1
+	index := notFound
+
+	for i, s := range didDoc.Service {
+		if s.Type == serviceType {
+			if index == notFound || didDoc.Service[index].Priority > s.Priority {
+				index = i
+			}
+		}
+	}
+
+	if index == notFound {
+		return nil, fmt.Errorf("service not found in DID document: %s", serviceType)
+	}
+
+	return &didDoc.Service[index], nil
+}
+
+// LookupRecipientKeys gets the recipient keys from the did doc which match the given parameters.
+func LookupRecipientKeys(didDoc *Doc, serviceType, keyType string) ([]string, error) {
```

Comment:
(later) we may want to think about refactoring these static accessors and the existing instance fields.